app/src/main/java/fr/neamar/kiss/SettingsActivity.java

```diff
@@ -147,23 +150,16 @@ public void onSharedPreferenceChanged(SharedPreferences sharedPreferences, Strin
             }
         }
 
-        if (requireRestartSettings.contains(key)) {
-            SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(this);
-            prefs.edit().putBoolean("require-layout-update", true).apply();
-
-            // Restart current activity to refresh view, since some
-            // preferences
-            // require using a new UI
-            Intent intent = new Intent(this, getClass());
-            intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP | Intent.FLAG_ACTIVITY_NEW_TASK
-                    | Intent.FLAG_ACTIVITY_NO_ANIMATION);
+        if(key.equalsIgnoreCase("theme")) {
+            requireFullRestart = true;
             finish();
-            overridePendingTransition(0, 0);
-            startActivity(intent);
-            overridePendingTransition(0, 0);
             return;
         }
 
+        if (requireRestartSettings.contains(key)) {
+            requireFullRestart = true;
+        }
```

Comment:
[PEDANTIC] Why not leave "theme" in the `requireRestartSettings` list and move the check for calling `this.finish()` after this `requireFullRestart = true;` assignment? IMHO it would make it clearer that changing the theme special only in the way that it is a are more "restarty" kind of restart than the other values.

```
if(requireRestartSettings.contains(key)) {
    this.requireFullRestart = true;

    // Theme changes also require settings to be reloaded to give the user instant feedback
    if(key.equalsIgnoreCase("theme")) {
        this.finish();
    }
}
```